src/14-curated-discovery/04-computer-vision.ts

```diff
@@ -0,0 +1,110 @@
+import dotenv from "dotenv"
+import _ from "lodash"
+import { describeImage, getArtworks, resizeImage } from "./helpers"
+import { program } from "commander"
+import { openai } from "@ai-sdk/openai"
+import { anthropic } from "@ai-sdk/anthropic"
+import chalk from "chalk"
+import { GravityArtwork } from "./types"
+import fs from "fs"
+import path from "path"
+
+dotenv.config()
+
+const DEFAULT_BATCH_SIZE = 10
+const OUTPUT_DIRECTORY = "./data/images"
+
+const { batchSize, model } = processCommandLineArguments()
+
+async function main() {
+  const artworks = await getArtworks()
+  const batches = _.chunk(artworks, batchSize)
+
+  console.log(
+    `Processing ${artworks.length} artworks in ${batches.length} batches with ${model.modelId}`
+  )
+
+  createOutputDirectory()
+
+  for (const batch of batches) {
+    await Promise.all(batch.map(processArtwork))
+  }
+  console.log("Done.")
+}
+
+async function processArtwork(artwork: GravityArtwork) {
+  if (isAlreadyDescribed(artwork)) {
+    console.log(chalk.gray.dim([artwork.id, artwork.slug].join(" ")))
+    return
+  }
+  try {
+    const resizedImageUrl = resizeImage(artwork.image_url, {
+      height: 512,
+      width: 512,
+    })
+    const description = await describeImage({
+      url: resizedImageUrl,
+      model,
+    })
+    const filePath = `${getOutputDirectory()}/${artwork.id}.json`
+    fs.writeFileSync(filePath, JSON.stringify(description.object, null, 2))
+    console.log(chalk.green([artwork.id, artwork.slug].join(" ")))
+  } catch (error) {
+    const filePath = `${getOutputDirectory()}/error-${artwork.id}.json`
+    fs.writeFileSync(filePath, JSON.stringify(error, null, 2))
+    console.log(chalk.red([artwork.id, artwork.slug].join(" ")))
+  }
+}
+
+function isAlreadyDescribed(artwork: GravityArtwork) {
+  const filePath = `${getOutputDirectory()}/${artwork.id}.json`
```

Comment:
As noted above we avoid re-describing artworks that have already been described. This is a measure to avoid spending time/money on re-processing, as tends to happen a lot during development. Basically just using the filesystem as cache, to keep things simple (though this prob won't scale once we index tens of thousands of artworks)